### ISSUE TITLE
[feature request] Support cc_import of .so with soname

### ISSUE THREAD

**Issue by yongtang** [bazelbuild/bazel#7059]
### Description of the problem / feature request:

At the moment `cc_import` supports linking to a shared library like `libmylib.so`. However, it does not support .so with soname, e.g., `libmylib.so.2`. It would be good to support .so with soname.

### Feature requests: what underlying problem are you trying to solve with this feature?

In linux .so with soname is very common so to expand the usage of `cc_import` to soname would be ideal.

**Comment by beasleyr-vmw** [comment]
Looks like this was touched on by 5efc8c473596f797445ecc4be7a0b0585654d8fc. Since that commit, one can do something like

```python
cc_import(
    name = "mylib",
    interface_library = "libmylib.so",  # ld searches for this w/ -lmylib
    shared_library = "libmylib.so.1",  # soname embedded in library; runtime linker searches for this
)
```

**Comment by c-mita** [comment]
As mentioned above, I believe this is resolved. If there's still an issue, please reopen.

**Comment by dfreese** [comment]
I ran into this as well, however, I was using `provided_by_system=True`, which means I'm limited to `interface_library`.  That field doesn't allow for a version to be specified.

Ran into this trying to provide a solution for `nvidia-encode.so`.  The nvidia docker runtime doesn't create the link from `nvidia-encode.so` to `nvidia-encode.so.1`.  As a result, there doesn't seem to be a great way to specify this in a way that bazel uses the system library when running tests, for example.

Let me know if this should be brought up as a separate issue (or should be considered an issue).
